linked_list/linked_list.py

```diff
@@ -84,45 +123,53 @@ def visit(self):
         
         print(", ".join(helper_list))
 
-    # method to reverse the singly linked list
-    # note: the nodes should be moved and not just the values in the nodes
-    # Time Complexity: ?
-    # Space Complexity: ?
+    # Time Complexity: O(n)
+    # Space Complexity: O(n)

```

Comment:
🪐 Space complexity would be O(1) here again because you aren't creating any additional data structures whose size is proportional to the size of your linked list